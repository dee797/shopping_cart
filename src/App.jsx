import { useState } from "react";
import { Link } from "react-router-dom";
import PropTypes from 'prop-types';

const App = () => {
  const [heading, setHeading] = useState("Magnificent Monkeys");

  const clickHandler = () => {
    setHeading("Radical Rhinos");
  };

  return (
    <>
      <div>
        <h1>Hello from the main page of the app!</h1>
        <p>Here are some examples of links to other pages</p>
        <nav>
          <ul>
            <li>
              <Link to="profile">Profile page</Link>
            </li>
          </ul>
        </nav>
      </div>

      { /* Use for Vitest/React Testing/Jest */ }
      <button type="button" onClick={clickHandler}>
        Click Me
      </button>
      <h1>{heading}</h1>
    </>
  );
};

// RenderName component
const RenderName = (props) => {
  return <div>{props.name}</div>;
};


// Set data types on RenderName props
RenderName.propTypes = {
  name: PropTypes.string.isRequired,
};

RenderName.defaultProps = {
  name: 'Zach',
};


export default App;