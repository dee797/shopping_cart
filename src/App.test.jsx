import { describe, it, expect } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";


describe("App component", () => {
  it("renders home page", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders shopping page", () => {
    render(<App />);
    const link = screen.getByTestId("shopLink");

    fireEvent.click(link);
    
    expect(screen.findByText(/items/i));

  });
});
