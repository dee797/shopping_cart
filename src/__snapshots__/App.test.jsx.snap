// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders home page 1`] = `
<div>
  <div
    class="container"
  >
    <nav>
      <ul
        class="flex"
      >
        <li>
          <a
            data-discover="true"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            data-discover="true"
            data-testid="shopLink"
            href="/shop"
          >
            Shop
          </a>
        </li>
      </ul>
    </nav>
    <h1>
      Welcome
    </h1>
    <p>
      This is not a real shop-- still, feel free to take a look around.
    </p>
  </div>
</div>
`;
